src/dotnet/Program.cs

```diff
@@ -178,6 +182,9 @@ private static void ConfigureDotNetForFirstTimeUse()
                             environmentProvider);
 
                         dotnetConfigurer.Configure();
+
+                        if (!File.Exists(Telemetry.TelemetrySentinel))
```

Comment:
The nugetcache sentinel is stored at nugetCachepath, and this path is calculated in some order of preference, starting from global.json to runtime dir packages folder. I don't see any logic for where it is being done user local. Can you point to the code ?